File: php/examples/BucketInventory.php
<?php

use Qcloud\Cos\Client;
use Qcloud\Cos\Exception\ServiceResponseException;

class BucketInventory
{
    private $cosClient;

    private $uploadId;
    private $eTag;
    private $versionId;

    // 设置存储桶清单任务
    protected function putBucketInventory() {
        $cosClient = $this->cosClient;
        //.cssg-snippet-body-start:[put-bucket-inventory]
        try {
            $result = $cosClient->putBucketInventory(array(
                'Bucket' => 'examplebucket-1250000000', //格式：BucketName-APPID
                'Id' => 'string',
                'Destination' => array(
                    'COSBucketDestination'=>array(
                        'Format' => 'CSV',
                        'AccountId' => '100000000001',
                        'Bucket' => 'qcs::cos:ap-chengdu::examplebucket-1250000000',
                        'Prefix' => 'string',
                    )
                ),      
                'IsEnabled' => 'True',
                'Schedule' => array(
                    'Frequency' => 'Daily',
                ),  
                'Filter' => array(
                    'Prefix' => 'string',
                ),  
                'IncludedObjectVersions' => 'Current',
                'OptionalFields' => array(
                    'Size', 
                    'ETag',
                )
            ));
            // 请求成功
            print_r($result);
        } catch (\Exception $e) {
            // 请求失败
            echo "$e\n";
        }
        
        
        //.cssg-snippet-body-end
    }

    // 获取存储桶清单任务
    protected function getBucketInventory() {
        $cosClient = $this->cosClient;
        //.cssg-snippet-body-start:[get-bucket-inventory]
        try {
            $result = $cosClient->getBucketInvnetory(array(
                'Bucket' => 'examplebucket-1250000000', //格式：BucketName-APPID
                'Id' => 'string',
            ));
            // 请求成功
            print_r($result);
        } catch (\Exception $e) {
            // 请求失败
            echo($e);
        }
        
        
        //.cssg-snippet-body-end
    }

    // 删除存储桶清单任务
    protected function deleteBucketInventory() {
        $cosClient = $this->cosClient;
        //.cssg-snippet-body-start:[delete-bucket-inventory]
        try {
            $result = $cosClient->deleteBucketInvnetory(array(
                'Bucket' => 'examplebucket-1250000000', //格式：BucketName-APPID
                'Id' => 'string',
            ));
            // 请求成功
            print_r($result);
        } catch (\Exception $e) {
            // 请求失败
            echo($e);
        }
        
        
        //.cssg-snippet-body-end
    }

	//.cssg-methods-pragma

    protected function init() {
        $secretId = "COS_SECRETID"; //"云 API 密钥 SecretId";
        $secretKey = "COS_SECRETKEY"; //"云 API 密钥 SecretKey";
        $region = "COS_REGION"; //设置一个默认的存储桶地域
        $this->cosClient = new Qcloud\Cos\Client(
            array(
                'region' => $region,
                'schema' => 'https', //协议头部，默认为http
                'credentials'=> array(
                    'secretId'  => $secretId ,
                    'secretKey' => $secretKey)));
    }

    public function mBucketInventory() {
        $this->init();

        // 设置存储桶清单任务
        $this->putBucketInventory();

        // 获取存储桶清单任务
        $this->getBucketInventory();

        // 删除存储桶清单任务
        $this->deleteBucketInventory();

	    //.cssg-methods-pragma
    }
}
?>

File: Python/examples/bucket_inventory.py
# -*- coding=utf-8

from qcloud_cos import CosConfig
from qcloud_cos import CosS3Client
from qcloud_cos import CosServiceError
from qcloud_cos import CosClientError

secret_id = 'COS_SECRETID'     # 替换为用户的secret_id
secret_key = 'COS_SECRETKEY'     # 替换为用户的secret_key
region = 'COS_REGION'    # 替换为用户的region
token = None               # 使用临时密钥需要传入Token，默认为空,可不填
config = CosConfig(Region=region, SecretId=secret_id, SecretKey=secret_key, Token=token)  # 获取配置对象
client = CosS3Client(config)

# 设置存储桶清单任务
def put_bucket_inventory():
    #.cssg-snippet-body-start:[put-bucket-inventory]
    response = client.put_bucket_inventory(
        Bucket='examplebucket-1250000000',
        Id='string',
        InventoryConfiguration={
            'Destination': {
                'COSBucketDestination': {
                    'AccountId': '100000000001',
                    'Bucket': 'qcs::cos:ap-guangzhou::examplebucket-1250000000',
                    'Format': 'CSV',
                    'Prefix': 'string',
                    'Encryption': {
                        'SSECOS': {}
                    }
                }
            },
            'IsEnabled': 'true'|'false',
            'Filter': {
                'Prefix': 'string'
            },
            'IncludedObjectVersions':'All'|'Current',
            'OptionalFields': {
                'Field': [
                    'Size',
                    'LastModifiedDate',
                    'ETag',
                    'StorageClass',
                    'IsMultipartUploaded',
                    'ReplicationStatus'
                ]
            },
            'Schedule': {
                'Frequency': 'Daily'|'Weekly'
            }
        }
    )
    
    #.cssg-snippet-body-end

# 获取存储桶清单任务
def get_bucket_inventory():
    #.cssg-snippet-body-start:[get-bucket-inventory]
    response = client.get_bucket_inventory(
        Bucket='examplebucket-1250000000',
        Id='string'
    )
    
    #.cssg-snippet-body-end

# 删除存储桶清单任务
def delete_bucket_inventory():
    #.cssg-snippet-body-start:[delete-bucket-inventory]
    response = client.delete_bucket_inventory(
        Bucket='examplebucket-1250000000',
        Id='string'
    )
    
    #.cssg-snippet-body-end

#.cssg-methods-pragma


# 设置存储桶清单任务
put_bucket_inventory()

# 获取存储桶清单任务
get_bucket_inventory()

# 删除存储桶清单任务
delete_bucket_inventory()

#.cssg-methods-pragma